File: app/src/main/java/com/example/builday365/viewmodel/SideBarViewModel.java
package com.example.builday365.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.builday365.model.Timeline.Memo;
import com.example.builday365.model.Timeline.TimeLine;

import java.sql.Time;
import java.util.Date;

public class SideBarViewModel extends ViewModel {

    private MutableLiveData<String> memo = new MutableLiveData<String>();
    private MutableLiveData<TimeLine> mtimeline = new MutableLiveData<TimeLine>();


    public SideBarViewModel() {
    }

    public LiveData<TimeLine> getTimeline() {
        return mtimeline;
    }
    // Method to add a memo
    public void addMemo(String memoContent, Date stime, Date etime) {

        TimeLine currentTimeLine = mtimeline.getValue();

        if(currentTimeLine == null)
        {
            currentTimeLine = new TimeLine();
        }

        Memo newMemo = new Memo(stime, etime, memoContent);
        currentTimeLine.getCurMemos().add(newMemo);
        mtimeline.setValue(currentTimeLine);
    }
}

File: app/src/main/java/com/example/builday365/viewmodel/SideBarViewModel.kt
package com.example.builday365.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.builday365.model.test.Memo
import com.example.builday365.model.test.TimeLine
import java.util.Date

class SideBarViewModel : ViewModel() {
    private val memo = MutableLiveData<String>()
    private val mtimeline = MutableLiveData<TimeLine>()
    var m_startTime_hr = 0
    val timeline: LiveData<TimeLine>
        get() = mtimeline

    // Method to add a memo
    fun addMemo(memoContent: String?, stime: Date?, etime: Date?) {
        var currentTimeLine = mtimeline.value
        if (currentTimeLine == null) {
            currentTimeLine = TimeLine()
        }
        val newMemo = Memo(stime, etime, memoContent)
        currentTimeLine.curMemos.add(newMemo)
        mtimeline.value = currentTimeLine
    }
}